app/models/bulk_tagging.rb

```diff
@@ -0,0 +1,25 @@
+require 'csv'
+
+class BulkTagging < ActiveRecord::Base
+  validates :mapping_url, presence: true
+  validate :https_url?
+  validate :public_google_sheet_url?
+
+  has_many :tag_mappings, dependent: :destroy
+
+  scope :newest_first, -> { order(created_at: :desc) }
+
+private
+
+  def https_url?
+    unless URI.parse(mapping_url).scheme == "https"
```

Comment:
Went in early on as a cursory validation that the mapping_url was actually a proper url. The validation below this one (checking that it matches the structure of a published google sheet url) is probably sufficient. Removing this one, it's just extra code with minimal value.